lib/api/hub_config_api.rb

```diff
@@ -35,15 +32,25 @@ def signing_certificates(entity_id)
 
 private
 
-  def encryption_cert_path(entity_id)
-    URI.join(HUB_HOST, CERTIFICATES_ROUTE, CERTIFICATE_ENCRYPTION_ENDPOINT % { entity_id: CGI.escape(entity_id) }).to_s
+  def use_secure_header(environment)
+    hub_environment(environment, 'secure-header') == 'true' && environment == 'integration'
+  end
+
+  def build_request(environment:, url:)
+    return Faraday.get(url) unless use_secure_header(environment)
+
+    Faraday.get(url) { |req| req.headers['X-Self-Service-Authentication'] = Rails.configuration.authentication_header }
+  end
+
+  def encryption_cert_path(environment, entity_id)
+    { environment: environment, url: URI.join(hub_environment(environment, 'hub-config-host'), CERTIFICATES_ROUTE, CERTIFICATE_ENCRYPTION_ENDPOINT % { entity_id: CGI.escape(entity_id) }).to_s }
```

Comment:
don't you want to rename `hub-config-host` to `hub_config_host`? Then you can do `hub_environment(environment, :hub_config_host)` and be consistent when doing `hub_environment(environment, :bucket)` , i.e. the key is always a symbol instead of sometimes being a symbol and sometimes a string

Reply:
I am inclined to meet you half way on this one, given that terraform seemed to use this format. I can however revert back to symbol form :'hub_config_host' which works. Rosa thought the string form looked better

Reply:
terraform is not involved at all. It's just a value in JSON of the environment variable, it just happens it's created in terraform. Then I would go with a string then. I'm not liking the inconsistency...

Reply:
@jakubmiarka @jkunle after our discussion earlier I have pushed a new commit to self-service to use symbols and have renamed the keys  in `verify-infrastructure-config` https://github.com/alphagov/verify-infrastructure-config/pull/235